menu/views.py

```diff
@@ -163,18 +163,23 @@ def suggestedcontent(request):
 
     suggested_pages = []
     for topic_id in topics_viewed:
-        for page in (
-            ContentPage.objects.get(id=topic_id)
-            .get_descendants()
-            .filter(numchild=0)
-            .live()
-        ):
+        try:
+            pages = (
+                ContentPage.objects.get(id=topic_id)
+                .get_descendants()
+                .filter(numchild=0)
+                .live()
+            )
+        except ContentPage.DoesNotExist:
+            pages = None
+
+        for page in pages:
```

Comment:
This will raise an error when pages in None:
`TypeError: 'NoneType' object is not iterable`

We could add a test for that. To fix you can make line 174 : `pages = []`